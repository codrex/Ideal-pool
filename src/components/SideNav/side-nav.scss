@import '../../style/_var.scss';
.side-nav {
  display: flex;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
  width: 200px;
  background: $primary-color;
  align-items: center;
  padding: 2.5rem;

  & hr {
    background-color: #fff;
    opacity: 0.4;
    width: 100%;
    height: 1px;
    border: none;
    margin: 2.8rem 0;
  }

  &__avatar .avatar__text {
    font-size: 1.2rem;
  }
}
