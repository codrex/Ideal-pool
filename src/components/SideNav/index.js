// @flow
import React, { PureComponent } from 'react';
import IdeaPoolIcon from '../../assets/images/IdeaPool_icon.png';
import Avatar from '../Avatar';
import './side-nav.scss';

type UserInfo = {
  name: string,
  avatarUrl: string,
};

type Props = {
  userInfo: UserInfo,
  isAuthenticated: boolean,
};
class SideNav extends PureComponent<Props> {
  static AppIcon() {
    return (
      <Avatar text="The Idea Pool">
        <img src={IdeaPoolIcon} alt="Idea Pool icon" />
      </Avatar>
    );
  }

  static UserAvatar(props: UserInfo) {
    const { name, avatarUrl } = props;
    return (
      <Avatar text={name} className="side-nav__avatar">
        <img src={avatarUrl} alt="user avatar" />
      </Avatar>
    );
  }

  render() {
    const { userInfo, isAuthenticated } = this.props;
    return (
      <div className="side-nav">
        <SideNav.AppIcon />
        <hr />
        {isAuthenticated && <SideNav.UserAvatar {...userInfo} />}
      </div>
    );
  }
}

export default SideNav;
