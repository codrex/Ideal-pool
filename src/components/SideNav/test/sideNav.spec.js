import React from 'react';
import { shallow } from 'enzyme';
import SideNav from '..';

const props = {
  userInfo: {
    name: 'Joyce Lee',
    avatarUrl: '',
  },
  isAuthenticated: true,
};
describe('SideNav', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SideNav {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render UserAvatar when user is authenticated', () => {
    const wrapper = shallow(<SideNav {...props} />);
    expect(wrapper.find('UserAvatar').length).toBe(1);
  });

  it('should not render UserAvatar when user is not authenticated', () => {
    const wrapper = shallow(<SideNav {...props} isAuthenticated={false} />);
    expect(wrapper.find('UserAvatar').length).toBe(0);
  });
});
