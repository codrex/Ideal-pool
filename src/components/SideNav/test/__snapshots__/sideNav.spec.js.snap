// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should match snapshot 1`] = `
<div
  className="side-nav"
>
  <AppIcon />
  <hr />
  <UserAvatar
    avatarUrl=""
    name="Joyce Lee"
  />
</div>
`;

exports[`SideNav should match snapshot when user is authenticated 1`] = `
<div
  className="side-nav"
>
  <AppIcon />
  <hr />
  <UserAvatar
    avatarUrl=""
    name="Joyce Lee"
  />
</div>
`;

exports[`SideNav should match snapshot when user is not authenticated 1`] = `
<div
  className="side-nav"
>
  <AppIcon />
  <hr />
</div>
`;
